parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoSchemaConverter.java

```diff
@@ -97,6 +127,46 @@ public MessageType convert(Class<? extends Message> protobufClass) {
 
   private <T> Builder<? extends Builder<?, GroupBuilder<T>>, GroupBuilder<T>> addField(FieldDescriptor descriptor, final GroupBuilder<T> builder) {
     if (descriptor.getJavaType() == JavaType.MESSAGE) {
+      if (unwrapProtoWrappers) {
+        String typeName = descriptor.getMessageType().getFullName();
+        if (typeName.equals(PROTOBUF_TIMESTAMP_TYPE)) {
+          return builder.primitive(INT64, getRepetition(descriptor)).as(timestampType(true, TimeUnit.NANOS));
```

Comment:
should TimeUnit be something configurable?  I know protobuf timestamps support nanosecond resolution but might not be what applications want to store.

Reply:
I don't think it's worth complicating the API. The Timestamp common Proto stores time in nanoseconds. There's no good reason to deviate from that or to truncate the resolution. If the user wishes to do more manipulation, it can be done downstream.

Reply:
In that case, you can use the default 'TimeUnit.NANOS' while having that configured.

Reply:
Isn't that outside the scope of ProtoSchemaConverter/ParquetWriter? I don't think it should go into the business of doing transformations. If the point of `ProtoSchemaConverter` is to give the closest representation of the Protobuf object in Parquet, then it should be NANOS and nothing more.

Reply:
@emkornfield  Do you still have a comment on this?